### PR TITLE
828 steamline win level info message

### PR DESCRIPTION
**Pull request by pmarsh-scottlogic**
## Description

Before, when the user would send a message that would cause the level to be won, we would check if `wonLevel` in the backend's response was `true`, and if so, we would add a congratulations message to the chat History, causing an extra API call.

Now, we return the congratulations messagge (`wonLevelMessage`) in the chat response. The front end simply appends it to the history if it is there, and can use this to determine if the user has won the level.

## Screenshot

image

## Notes
also does these things:
- renames the `chatMessageType` `LEVEL_INFO` to `LEVEL_COMPLETE`
- removes the information header from the congratulations message. Adds party poppers instead

## Concerns

- Now in the backend, the type `ChatHttpResponse` contains the `wonLevel` boolean property. It gets marked as true at the points of the send email function call. At the end of `handleChatToGPT` in the _chatController_, we check `wonLevel`. If we have won the level, then we add the congratulations message to the history and return it to the frontend. The frontend only needs the congratualtions message to determine that a level has been won, making wonLevel redundant in the frontend. This is untidy, but not necessarily problematic. This will be made irrelevant by #780 

## Checklist

Have you done the following?

- [x] Linked the relevant Issue
- [x] Added tests
- [x] Ensured the workflow steps are passing

### CODE REVIEW

File: frontend/src/components/ChatBox/ChatBox.tsx [hunk 1]
```diff
@@ -152,19 +141,13 @@ function ChatBox({
...
-				type: 'LEVEL_INFO',
-				message: successMessage,
-			});
-			// asynchronously add the message to the chat history
-			void chatService.addInfoMessageToChatHistory(
-				successMessage,
-				'LEVEL_INFO',
-				currentLevel
+			const levelCompleteMessage = makeChatMessageFromDTO(
+				response.wonLevelMessage
```
Comment (chriswilty) [inline frontend/src/components/ChatBox/ChatBox.tsx:147]: This is maybe something that will emerge after all your changes, but my feeling is that it's fine to return the "wonLevel" flag as your indicator, and the actual "won level" message could arrive as an info message.

Reason being, we already have an optional info message in the response that is added to the chat panel ("transformedMessage" though it's currently a string, not an actual ChatMessage), and so adding another optional field for another info message to the response seems like we are making our API a little too specific. Maybe we want to return a list of (zero or more) info messages to add to the chat?

Likely not something to tackle here, but something to consider as you continue with this work.

File: frontend/src/components/ChatBox/ChatBoxMessage/MessageBubble.tsx [hunk 2]
```diff
@@ -48,12 +48,6 @@ function MessageBubble({
 	return (
 		// eslint-disable-next-line jsx-a11y/no-noninteractive-tabindex
 		<section className={className} lang="en" tabIndex={0}>
-			{message.type === 'LEVEL_INFO' && (
-				<>
-					<p className="header">Information</p>
-					<span className="visually-hidden"> message: </span>
-				</>
```
Comment (chriswilty) [inline frontend/src/components/ChatBox/ChatBoxMessage/MessageBubble.tsx:55]: Yeah that seems redundant, happy for it to go.

**Review by chriswilty — CHANGES_REQUESTED** [review]
Minor suggestions only, and one to chew on.

File: frontend/src/components/ChatBox/ChatBox.tsx (hunk 1 above)
Comment (pmarsh-scottlogic) [reply frontend/src/components/ChatBox/ChatBox.tsx:146]: I like the suggestion. Like, what happens if the message gets transformed, but that message would also win the level? My current solution doesn't support that

Comment (pmarsh-scottlogic) [reply frontend/src/components/ChatBox/ChatBox.tsx:146]: nevermind, yes it does. But your idea is sleeker

Comment (pmarsh-scottlogic) [reply frontend/src/components/ChatBox/ChatBox.tsx:146]: that can be a different PR :)

**Review by chriswilty — APPROVED** [review]
Lovely!

File: frontend/src/components/ChatBox/ChatBox.tsx (hunk 1 above)
Comment (pmarsh-scottlogic) [reply frontend/src/components/ChatBox/ChatBox.tsx:146]: https://github.com/orgs/ScottLogic/projects/37/views/2?pane=issue&itemId=57715194 made a ticket for it

**Comment by pmarsh-scottlogic** [comment]
I need to solve a problem where we are not sending the congratulations message if there is an OpenAI error after the winning email has been sent

**Comment by pmarsh-scottlogic** [comment]
I've decided the gain is not worth the cost. In the startlingly unlikely event that the user would win a level with a sent email, but open has an error after the function call, then the user will not win the level.

**Pull request merged**